File: codes/0413-Arithmetic-Slices/Java/main3.java
class Solution {
    public int numberOfArithmeticSlices(int[] A) {
        int res = 0, cur = 0;
        for (int i = 2; i < A.length; ++i) {
            if (A[i] - A[i - 1] == A[i - 1] - A[i - 2]) {
                cur += 1;
                res += cur;
            } else {
                cur = 0;
            }
        }
        return res;
    }
}

File: 剑指 Offer/剑指 Offer 06. 从尾到头打印链表/cpp/main3.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution
{
public:
    vector<int> reversePrint(ListNode *head)
    {
        vector<int> res;
        stack<int> s;
        ListNode *p = head;
        while (p)
        {
            s.push(p->val);
            p = p->next;
        }
        while (!s.empty())
        {
            res.push_back(s.top());
            s.pop();
        }
        return res;
    }
};
